src/mixins/createFormFieldMixin.ts

```diff
@@ -157,25 +249,42 @@ const createFormMixin: FormMixinFactory = createStateful
 				}
 			},
 
-			nodeAttributes: [
-				function (this: FormFieldMixin<any, FormFieldMixinState<any>>): VNodeProperties {
-					const { type, value, state } = this;
-					const { disabled, name } = state;
+			getNode(this: Widget<WidgetState, WidgetProperties> & FormField<any>): DNode {
+				const { label } = this;
+				let tag = label ? 'label' : 'div';
 
-					return { type, value, name, disabled: Boolean(disabled) };
+				if (this.classes.length) {
+					tag = `${this.tagName}.${this.classes.join('.')}`;
 				}
-			]
+
+				return v(tag, this.getNodeAttributes(), generateWrappedChildren.call(this));
+			}
 		},
 		initialize(
 			instance: FormFieldMixin<any, FormFieldMixinState<any>>,
-			{ value, type }: FormFieldMixinOptions<any, FormFieldMixinState<any>> = {}
+			{ value, type, properties }: FormFieldMixinOptions<any, FormFieldMixinState<any>> & WidgetOptions<WidgetState, FormFieldMixinProperties> = {}
 		) {
 			if (value) {
 				instance.setState({ value });
 			}
 			if (type) {
 				instance.type = type;
 			}
+			if (properties && properties.hasOwnProperty('label')) {
+				const labelDefaults = {
+					content: '',
+					position: 'below',
+					hidden: false
+				};
+
+				// convert string label to object
+				if (typeof properties['label'] === 'string') {
+					instance.label = assign(labelDefaults, { content: properties['label'] });
```

Comment:
why do we need to put this on the instance and not just use from properties during the render cycle?